src/dimension_slice.c

```diff
@@ -685,7 +711,7 @@ ts_dimension_slice_insert_multi(DimensionSlice **slices, Size num_slices, bool o
 		if (only_non_existing)
 		{
 			slices[i]->fd.id = 0;
-			slices[i] = ts_dimension_slice_scan_for_existing(slices[i]);
+			ts_dimension_slice_scan_for_existing_and_lock_found(slices[i], LockTupleKeyShare);
```

Comment:
I suggest adding a comment here on why tuple locking is needed, and why KEY SHARE is the appropriate lock mode.